coremltools/models/model.py

```diff
@@ -363,6 +363,8 @@ def __del__(self):
         # Cleanup temporary package upon destruction
         if hasattr(self, 'is_package') and self.is_package \
            and hasattr(self, 'is_temp_package') and self.is_temp_package:
+            if _shutil is None:
+                import shutil as _shutil
             _shutil.rmtree(self.package_path)
```

Comment:
isn't the issue in the line `_shutil.rmtree(self.package_path)` ? if the `self.package_path` is None or empty and hence there is nothing to remove?

Reply:
It is because the package itself (shutil) is destructed before this __del__ function is called